modules/internal/ChapelBase.chpl

```diff
@@ -840,8 +841,14 @@ module ChapelBase {
   // locale as the sync/cofall/cobegin was initiated on and thus the
   // same locale on which the object is allocated.
   pragma "dont disable remote value forwarding"
-  inline proc _endCountAlloc() {
-    return new _EndCount();
+  inline proc _endCountAlloc(param forceLocalTypes : bool) {
```

Comment:
I think we want a network atomic `taskCnt` whenever the _EndCount object is visible to another locale because of the code in _upEndCount. I was under the impression that this would mean we could use the same argument for both `i` and `taskCnt`. Is that accurate?

I'm tempted to not have a default value for `forceLocalTypes` to make sure that a developer has to think about this issue if they modify _EndCount code in the future. Otherwise, I think it could be easy to forget about. I'm open to making it a default though.